conda_build/source.py

```diff
@@ -347,7 +352,7 @@ def git_mirror_checkout_recursive(
             print(f"checkout: {checkout!r}")
         if checkout:
             check_call_env(
-                [git, "checkout", checkout],
+                [git, "checkout", "-f", checkout],  # UNDO
```

Comment:
This is needed to avoid a weird `git checkout` error with the `conda-build=1.8.1` clone test. Not sure why it detects that the repo has changes, possibly introduced by another test somewhere else. I did try to change the git clone cache but still failed. I'll try to take another look but UUUUGH conda-build.

Reply:
Gave up and switched to constructor for this test.